client/src/common/components/Comments/CommentsList.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import _map from 'lodash/map';
+import _isEmpty from 'lodash/isEmpty';
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+
+import Comment from 'common/components/Comments/Comment';
+import MessageBox from 'common/components/MessageBox';
+import { MessageType } from 'common/constants/enums';
+import NewComment from 'common/components/Comments/NewComment';
+import {
+  addComment,
+  fetchComments
+} from 'modules/list/components/Items/model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import Preloader from 'common/components/Preloader';
+
+class CommentsList extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    isNewCommentVisible: false,
+    pending: false
+  };
+
+  componentDidMount() {
+    const {
+      fetchComments,
+      itemId,
+      match: {
+        params: { id: listId }
+      }
+    } = this.props;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(fetchComments(listId, itemId));
+    this.pendingPromise.promise
+      .then(() => this.setState({ pending: false }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          this.setState({ pending: false });
+        }
+      });
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  showAddComment = () => this.setState({ isNewCommentVisible: true });
+
+  hideAddComment = () => this.setState({ isNewCommentVisible: false });
+
+  handleAddComment = comment => {
+    const {
+      addComment,
+      itemId,
+      match: {
+        params: { id: listId }
+      }
+    } = this.props;
+    const action = addComment(listId, itemId, comment);
+
+    action.then(() => this.hideAddComment());
+
+    return action;
+  };
```

Comment:
those 3 lines could be:
```
return addComment(listId, itemId,comment).then(() => this.hideAddComment());
```

Reply:
In the end, it probably would work the same. But here is the difference: I think external "then" should depend on adding a comment to the DB, not on hiding form and it would be the other way around if I implement this proposal.